components/atom/button/src/index.js

```diff
@@ -0,0 +1,98 @@
+import React, {PropTypes} from 'react'
+import cx from 'classnames'
+
+const CLASS = 'sui-AtomButton'
+const TYPES = ['primary', 'accent', 'secondary', 'tertiary']
+const MODIFIERS = ['disabled', 'small', 'large', 'fullWidth', 'focused']
+
+const appendClass = (suffix) => `${CLASS}--${suffix}`
+const includes = (array, item) => array.indexOf(item) !== -1
+const cleanProps = (props, Component) => {
+  let newProps = {...props}
+  Object.keys(Component.propTypes).forEach(key => delete newProps[key])
+  return newProps
+}
+
+const getTypes = (props) => {
+  let types = Object.keys(props)
+    .filter(name => includes(TYPES, name))
+  return types.length ? types : ['primary']
+}
+const getModifiers = (props) => {
+  return Object.keys(props)
+    .filter(name => includes(MODIFIERS, name))
+}
+
+const AtomButton = (props) => {
+  const {disabled, leftIcon, rightIcon, children, className} = props
+  const classNames = cx(
+    CLASS,
+    getTypes(props).map(appendClass),
+    getModifiers(props).map(appendClass),
+    className
+  )
+  const newProps = cleanProps(props, AtomButton)
+  return (<button {...newProps} className={classNames} disabled={disabled}>
+    {leftIcon && <span className={`${CLASS}LeftIcon`}>{leftIcon}</span>}
+    {children}
+    {rightIcon && <span className={`${CLASS}RightIcon`}>{rightIcon}</span>}
+  </button>)
+}
+
+AtomButton.displayName = 'AtomButton'
+AtomButton.propTypes = {
+  /**
```

Comment:
I chose an API based on aggregable props. button-basic has a different approach "type", "color", ...
Important to give your opinion as we can define new conventions to make our set of components more consistent.